### PR TITLE
docs: add pruning section

### PR DESCRIPTION
**Pull request by tac0turtle**
- added a pruning section to joinmainnet section.

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]

File: docs/gaia-tutorials/join-mainnet.md [hunk 1]
```diff
@@ -111,7 +111,19 @@ For mainnet, the recommended `gas-prices` is `0.025uatom`.
...
-The initial recommended `min-gas-prices` is `0.025uatom`, but you might want to change it later. 
+The initial recommended `min-gas-prices` is `0.025uatom`, but you might want to change it later.
+
+## Pruning of State
+
+There are three strategies for pruning state, please be aware that this is only for state and not for block storage:
+
+1. `PruneEverything`: This means that all saved states will be pruned other than the current.
+2. `PruneNothing`: This means that all state will be saved and nothing will be deleted.
+3. `PruneSyncable`: This means that only the state of the last 100 and every 10,000th will be saved.
```
Comment (fedekunze) [inline docs/gaia-tutorials/join-mainnet.md]:
```suggestion
3. `PruneSyncable`: This means that only the state of the last 100 and every 10,000th blocks will be saved.
```

File: docs/gaia-tutorials/join-mainnet.md [hunk 2]
```diff
@@ -111,7 +111,19 @@ For mainnet, the recommended `gas-prices` is `0.025uatom`.
...
+
+By default every node is in `PruneSyncable` mode. If you would like to change your nodes pruning strategy then you must do so when the node is initialized. For example, if you would like to change your node to the `PruneEverything` mode then you can pass the `---pruning everything` flag when you call `gaiad start`. If
```
Comment (fedekunze) [inline docs/gaia-tutorials/join-mainnet.md]: If ...

**Review by fedekunze — CHANGES_REQUESTED** [review]
Please complete the `If...`

File: docs/gaia-tutorials/join-mainnet.md (hunk 2 above)
Comment (tac0turtle) [reply docs/gaia-tutorials/join-mainnet.md]: not sure if was meant for here or somewhere else and I forgot to click into that window

**Review by alexanderbez — COMMENTED** [review]
I would wait until we have the pruning fix merged onto SDK `master` and we'll want to make note of the new changes to flushing to disk.

**Review by fedekunze — APPROVED** [review]

**Comment by tac0turtle** [comment]
I based this addition off of 0.37.6 of the sdk since it is what is used on the hub. Once the pruning is figured out in 0.38 i will make a section in the docs of the sdk with the new changes

**Pull request merged**